sycl/source/detail/scheduler/commands.cpp

```diff
@@ -3011,6 +3023,108 @@ ur_result_t ExecCGCommand::enqueueImpCommandBuffer() {
+    // using cuGraphAddChildGraphNode/hipGraphAddChildGraphNode.
+    //
+    // As both the SYCL-RT and UR adapter need to have a handle
+    // to this child command-buffer, the SYCL-RT creates it and
+    // then passes the handle via a parameter to
+    // urCommandBufferAppendNativeCommandExp.
+    ur_bool_t DeviceHasSubgraphSupport = false;
+    Adapter->call<UrApiKind::urDeviceGetInfo>(
+        DeviceImpl->getHandleRef(),
+        UR_DEVICE_INFO_COMMAND_BUFFER_SUBGRAPH_SUPPORT_EXP, sizeof(ur_bool_t),
+        &DeviceHasSubgraphSupport, nullptr);
+
+    ur_exp_command_buffer_handle_t ChildCommandBuffer = nullptr;
+    if (DeviceHasSubgraphSupport) {
+      ur_exp_command_buffer_desc_t Desc{
+          UR_STRUCTURE_TYPE_EXP_COMMAND_BUFFER_DESC /*stype*/,
+          nullptr /*pnext*/, false /* updatable */, false /* in-order */,
+          false /* profilable*/
+      };
+      Adapter->call<sycl::detail::UrApiKind::urCommandBufferCreateExp>(
+          ContextImpl->getHandleRef(), DeviceImpl->getHandleRef(), &Desc,
+          &ChildCommandBuffer);
+    }
+
+    CGHostTask *HostTask = (CGHostTask *)MCommandGroup.get();
+
+    // Extract the Mem Objects for all Requirements, to ensure they are
+    // available if a user asks for them inside the interop task scope
+    std::vector<interop_handle::ReqToMem> ReqToMem;
+    const std::vector<Requirement *> &HandlerReq = HostTask->getRequirements();
+    auto ReqToMemConv = [&ReqToMem, ContextImpl](Requirement *Req) {
+      const std::vector<AllocaCommandBase *> &AllocaCmds =
+          Req->MSYCLMemObj->MRecord->MAllocaCommands;
+
+      for (AllocaCommandBase *AllocaCmd : AllocaCmds)
+        if (ContextImpl == getContext(AllocaCmd->getQueue())) {
+          auto MemArg =
+              reinterpret_cast<ur_mem_handle_t>(AllocaCmd->getMemAllocation());
+          ReqToMem.emplace_back(std::make_pair(Req, MemArg));
+          return;
+        }
+
+      assert(false && "Can't get memory object due to no allocation available");
+
+      throw sycl::exception(
+          sycl::make_error_code(sycl::errc::runtime),
+          "Can't get memory object due to no allocation available " +
+              codeToString(UR_RESULT_ERROR_INVALID_MEM_OBJECT));
+    };
+    std::for_each(std::begin(HandlerReq), std::end(HandlerReq), ReqToMemConv);
+
+    ur_exp_command_buffer_handle_t InteropCommandBuffer =
+        ChildCommandBuffer ? ChildCommandBuffer : MCommandBuffer;
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+    // CMPLRLLVM-66082
+    // The native command-buffer should be a member of the sycl::interop_handle
+    // class, but it is in an ABI breaking change to add it. So member lives in
+    // the queue as a intermediate workaround.
+    interop_handle IH{ReqToMem, MQueue, DeviceImpl, ContextImpl,
+                      InteropCommandBuffer};
+#else
+    MQueue->setInteropGraph(InteropCommandBuffer);
+    interop_handle IH{ReqToMem, MQueue, DeviceImpl, ContextImpl};
+#endif
```

Comment:
Since the argument is now always there, we might as well reduce the amount of divergent code.
```suggestion
    interop_handle IH{ReqToMem, MQueue, DeviceImpl, ContextImpl,
                      InteropCommandBuffer};
#ifndef __INTEL_PREVIEW_BREAKING_CHANGES
    // CMPLRLLVM-66082
    // The native command-buffer should be a member of the sycl::interop_handle
    // class, but it is in an ABI breaking change to add it. So member lives in
    // the queue as a intermediate workaround.
    MQueue->setInteropGraph(InteropCommandBuffer);
#endif
```